vsintegration/src/FSharp.Editor/Common/Logger.fsi

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+
+// FIXME: We cannot make this internal yet until F# gets a compiler switch to make cases public when the type itself is internal.
+// https://github.com/Microsoft/visualfsharp/issues/4821
+type (* internal *) LogEditorFunctionId =
+    | SemanticClassification = 1
+    | SyntacticClassification = 2
+    | HandleCommandLineArgs = 3
+
+[<RequireQualifiedAccess>]
+module internal Logger =
+
+    val Log : LogEditorFunctionId -> unit
+
+    val LogBlockStart : LogEditorFunctionId -> unit
+
+    val LogBlockStop : LogEditorFunctionId -> unit
+
+    val LogBlock : LogEditorFunctionId -> IDisposable
```

Comment:
The problem with this is we create a trash object to hold the Disposable whether we are logging or not.  It would be preferable to manually add the BlockStart and BlockEnd inside an explicit try finally.

I realize that it's only one allocation, there are tons already in this code, however, we spend a ton of time to to reduce allocations, it would be nicer not to design more in.

Reply:
I've been thinking about this a lot more. I am ok with doing LogBlockStart and LogBlockStops inside an explicit try finally on items that do get called a lot (10s+ of thousands) in less than a second.

However, using the LogBlock as a default is very easy to use and does exactly what we want. I absolutely believe it is safe to use LogBlock despite allocations as long as you know how often a function gets called in a short amount of time. Even 100s to a 1000 times fine here.

I want to keep LogBlock as is. If we decide that it is a problem, we will be able to easily profile if it ever becomes one. At the moment, I don't see it as an issue.